packages/react-components/components/label/src/label.jsx

```diff
@@ -39,10 +52,13 @@ export function PureLabel({ naked, className, forwardedRef, children, ...props }
     const renderLabel = () => {
         const classes = mergeClasses(
             naked && "naked",
+            button && "with-button",
+            icon && "with-icon",
+            iconPosition === "right" && "with-icon-right",
             className
         );
 
-        return <SemanticLabel className={classes} {...props}>{children}</SemanticLabel>;
+        return <SemanticLabel className={classes} {...props}>{icon && iconPosition !== "right" ? icon : null}{children}{icon && iconPosition === "right" ? icon : null}</SemanticLabel>;
```

Comment:
This code is fine, however hard to understand and scale, I would prefer something similar to the following:

```
const renderLabelContent = () => {
    if (!isNil(icon)) {
         if (iconPosition === "right") {
              return <>{children}{icon}</>
         }

         return <>{icon}{children}</>
    }

    return children;
};

const renderLabel = () => {
    const classes = mergeClasses(...);

    return <SemanticLabel className={classes} {...props}>{renderLabelContent()}</SemanticLabel>
};
```

P.S. you might not need the fragments in `renderLabelContent`

Reply:
IMO, we should also add an "icon" class on the Icon component and in SUI theme we should style `.icon` instead of `svg`.